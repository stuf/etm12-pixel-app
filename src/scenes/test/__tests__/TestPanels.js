import * as React from 'karet';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import TestPanelsScene from 'scenes/test/TestPanels';

describe('TestPanelsScene', () => {
  it('renders and matches snapshot', () => {
    const wrapper = mount(<TestPanelsScene />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
