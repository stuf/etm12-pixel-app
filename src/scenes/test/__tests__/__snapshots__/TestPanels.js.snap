// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestPanelsScene renders and matches snapshot 1`] = `
<TestPanelsScene>
  <div
    className="scene-root"
  >
    <Panel>
      <div
        className="panelRoot vertical"
        style={
          Object {
            "gridTemplateRows": "30px auto 50px",
          }
        }
      >
        <div
          className="panelRoot__children"
        >
          <Panel
            direction="horizontal"
            size={30}
          >
            <div
              className="panelRoot horizontal"
              style={
                Object {
                  "gridTemplateColumns": "auto",
                }
              }
            >
              <div
                className="panelRoot__children"
              >
                main top
              </div>
            </div>
          </Panel>
          <Panel
            direction="horizontal"
          >
            <div
              className="panelRoot horizontal"
              style={
                Object {
                  "gridTemplateColumns": "200px auto 200px",
                }
              }
            >
              <div
                className="panelRoot__children"
              >
                <Panel
                  size={200}
                >
                  <div
                    className="panelRoot vertical"
                    style={
                      Object {
                        "gridTemplateRows": "50px auto",
                      }
                    }
                  >
                    <div
                      className="panelRoot__children"
                    >
                      <Panel
                        size={50}
                      >
                        <div
                          className="panelRoot vertical"
                          style={
                            Object {
                              "gridTemplateRows": "auto",
                            }
                          }
                        >
                          <div
                            className="panelRoot__children"
                          >
                            left top (50px)
                          </div>
                        </div>
                      </Panel>
                      <Panel>
                        <div
                          className="panelRoot vertical"
                          style={
                            Object {
                              "gridTemplateRows": "auto",
                            }
                          }
                        >
                          <div
                            className="panelRoot__children"
                          >
                            left bottom
                          </div>
                        </div>
                      </Panel>
                    </div>
                  </div>
                </Panel>
                <Panel>
                  <div
                    className="panelRoot vertical"
                    style={
                      Object {
                        "gridTemplateRows": "auto",
                      }
                    }
                  >
                    <div
                      className="panelRoot__children"
                    >
                      center auto width
                    </div>
                  </div>
                </Panel>
                <Panel
                  size={200}
                >
                  <div
                    className="panelRoot vertical"
                    style={
                      Object {
                        "gridTemplateRows": "auto 100px",
                      }
                    }
                  >
                    <div
                      className="panelRoot__children"
                    >
                      <Panel>
                        <div
                          className="panelRoot vertical"
                          style={
                            Object {
                              "gridTemplateRows": "auto",
                            }
                          }
                        >
                          <div
                            className="panelRoot__children"
                          >
                            right top
                          </div>
                        </div>
                      </Panel>
                      <Panel
                        size={100}
                      >
                        <div
                          className="panelRoot vertical"
                          style={
                            Object {
                              "gridTemplateRows": "auto",
                            }
                          }
                        >
                          <div
                            className="panelRoot__children"
                          >
                            right bottom (100px)
                          </div>
                        </div>
                      </Panel>
                    </div>
                  </div>
                </Panel>
              </div>
            </div>
          </Panel>
          <Panel
            direction="horizontal"
            size={50}
          >
            <div
              className="panelRoot horizontal"
              style={
                Object {
                  "gridTemplateColumns": "auto",
                }
              }
            >
              <div
                className="panelRoot__children"
              >
                bottom fullwidth (50px)
              </div>
            </div>
          </Panel>
        </div>
      </div>
    </Panel>
  </div>
</TestPanelsScene>
`;
